pkg/alertmanager/multitenant.go

```diff
@@ -737,6 +766,59 @@ func (am *MultitenantAlertmanager) ServeHTTP(w http.ResponseWriter, req *http.Re
 		return
 	}
 
+	if am.cfg.ShardingEnabled && am.distributor.IsPathSupported(req.URL.Path) {
```

Comment:
Concern: once this PR gets merged, the distributor will be always used if sharding is enabled. Shouldn't we introduce a temporarily flag to enable the distributor (disabled by default) so that we'll have control over it while testing / rolling out?

Reply:
Discussed offline, not required